final_project/perturb.py

```diff
@@ -0,0 +1,332 @@
+"""
+APC 524 
+Final Project
+Quantifying resilience of coupled networks
+
+@author Jalal Butt
+Date: 12-03-22
+
+
+Module file handling the perturbation network
+----------------------------------------------
+
+
+Outputs:
+--------
+	- time-evolution of perturbation
+		--> entirely independent of what's happening on the grid. I.e., need not output 
+			individual time-steps
+"""
+
+
+
+## (Elliptic Equations hw)
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy.sparse as sparse
+import scipy.sparse.linalg as linalg
+# from scipy import signal
+
+from dataclasses import dataclass
+import traceback
+
+
+## Arb parameters for testing.
+
+## external inputs, to be fed in from 
+"""
+m = 100 # number of grid-points
+n = 100
+n = 100 #  currently not implemented
+L_x = 10
+H_y = 10
+
+f_type = 'oscillatory'
+f_type = 'delta'
+source_point = (-2,-2)
+source_strength = 1000
+radius = 10
+"""
```

Comment:
is this docstring needed?

Reply:
nope, but I'm using it as a personal running reference of inputs. 

separately, 1.) you said merging but you're really just rebasing with jb_perturb_1, right? 
2.) for your attempt-run note the "example usage" for input parameters; not all have assigned defaults.